src/kbmod/fake_data_creator.py

```diff
@@ -0,0 +1,206 @@
+                    self.height,
+                    self.noise_level,
+                    self.noise_level**2,
+                    self.times[i],
+                    p,
+                )
+            image_list.append(img)
+
+        stack = image_stack(image_list)
+        return stack
+
+    def insert_object(self, trj):
+        """Insert a fake object given the trajectory.
+
+        Parameters
+        ----------
+        trj : trajectory
+            The trajectory of the fake object to insert.
+        """
+        img_count = len(self.times)
+        t0 = self.times[0]
+
+        for i in range(img_count):
+            dt = self.times[i] - t0
+            px = trj.x + dt * trj.x_v + 0.5
+            py = trj.y + dt * trj.y_v + 0.5
+
+            # Get the image for the timestep, add the object, and
+            # re-set the image. This last step needs to be done
+            # explicitly because of how pybind handles references.
+            current_layered_image = self.stack.get_single_image(i)
+            current_layered_image.add_object(px, py, trj.flux)
+            self.stack.set_single_image(i, current_layered_image)
+
+        # Save the trajectory into the internal list.
+        self.trajectories.append(trj)
+
+    def insert_random_object(self, flux):
+        """Create a fake object and insert it into the image.
+
+        Parameters
+        ----------
+        flux : float
+            The flux of the object.
+
+        Returns
+        -------
+        t : trajectory
+            The trajectory of the inserted object.
+        """
+        dt = self.times[-1] - self.times[0]
+
+        # Create the random trajectory.
+        t = trajectory()
+        t.x = int(random.random() * self.width)
+        xe = int(random.random() * self.width)
+        t.x_v = (xe - t.x) / dt
+        t.y = int(random.random() * self.height)
+        ye = int(random.random() * self.height)
+        t.y_v = (ye - t.y) / dt
+        t.flux = flux
+
+        # Insert the object.
+        self.insert_object(t)
+
+        return t
+
+    def save_fake_data(self, data_dir):
+        """Create the fake data in a given directory.
+
+        Parameters
+        ----------
+        data_dir : str
+            The path of the directory for the fake data.
+        """
+        # Make the subdirectory if needed.
+        dir_path = Path(data_dir)
+        if not dir_path.is_dir():
+            print("Directory '%s' does not exist. Creating." % data_dir)
+            os.mkdir(data_dir)
+
+        # Save each of the image files.
+        for i in range(self.stack.img_count()):
+            img = self.stack.get_single_image(i)
+            filename = f"{dir_path}/{img.get_name()}.fits"
+
+            # If the file already exists, delete it.
+            if Path(filename).exists():
+                os.remove(filename)
+
+            # Save the file.
+            img.save_layers(data_dir + "/")
+
+            # Open the file and insert fake WCS data.
```

Comment:
Would any of these numbers ever need to change? Alternatively if these number change, will it break unit tests? If so a comment about that would be helpful.

Reply:
I could see some instances where someone would want to change these, but it would require much of the set to be changed together. Right now the goal is to have some valid values so KBMOD can load the WCS. So I think we are okay using fixed values for now.